src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -39,7 +39,7 @@ public void Apply(BsonClassMap classMap)
                 return;
             }
 
-            var properties = typeInfo.GetProperties();
+            var properties = typeInfo.GetProperties(BindingFlags.Instance | BindingFlags.Public );
```

Comment:
this change affects two code paths:
1. https://github.com/sindbach/mongo-csharp-driver/pull/3/files#diff-1bcd6b847860a1520a8a7ea2b958170aR44
2. https://github.com/sindbach/mongo-csharp-driver/pull/3/files#diff-1bcd6b847860a1520a8a7ea2b958170aR53

But the created test validated only the second one. Should we add a test to check the first?